### PR TITLE
added image processing class to preprocess the image before running OCR

### PR DESCRIPTION
**Pull request by ayabel**
## Change Description

Added an option to run an image preprocessor in the image analyzer engine

## Issue reference

This PR fixes issue #XX

## Checklist

- [x ] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [ ] I have signed the CLA (if required)
- [x ] My code includes unit tests
- [x ] All unit tests and lint checks pass locally
- [x ] My PR contains documentation updates / additions if required

### CODE REVIEW

File: presidio-image-redactor/presidio_image_redactor/image_processing_engine.py
Comment (niwilso) [inline presidio-image-redactor/presidio_image_redactor/image_processing_engine.py:106]: Would you mind adding more detail in the docstring for the class, just to provide an overview of what this class is/does?

File: presidio-image-redactor/presidio_image_redactor/image_processing_engine.py
Comment (niwilso) [inline presidio-image-redactor/presidio_image_redactor/image_processing_engine.py:149]: Please add an overview of what this class does in the docstring.

File: presidio-image-redactor/presidio_image_redactor/image_processing_engine.py
Comment (niwilso) [inline presidio-image-redactor/presidio_image_redactor/image_processing_engine.py:153]: Please add type hinting for the arguments in this init method.

File: presidio-image-redactor/presidio_image_redactor/image_processing_engine.py
Comment (niwilso) [inline presidio-image-redactor/presidio_image_redactor/image_processing_engine.py:222]: Please add type hinting for the arguments in this init method.

File: presidio-image-redactor/presidio_image_redactor/image_processing_engine.py
Comment (niwilso) [inline presidio-image-redactor/presidio_image_redactor/image_processing_engine.py:268]: Please provide a basic overview of what this approach is in the docstring.

File: presidio-image-redactor/presidio_image_redactor/image_processing_engine.py
Comment (niwilso) [inline presidio-image-redactor/presidio_image_redactor/image_processing_engine.py:276]: For consistency, please either use `-> None` or don't for all your no-return methods in this file.

**Review by niwilso — CHANGES_REQUESTED** [review]
Overall, this looks great! Just left some comments to be addressed before approval.

**Comment by omri374** [comment]
@ayabel looks like the e2e tests are failing. Have you looked into this? Is it because the new logic changes the output?

File: presidio-image-redactor/presidio_image_redactor/image_analyzer_engine.py
Comment (ayabel) [reply presidio-image-redactor/presidio_image_redactor/image_analyzer_engine.py:56]: thanks! @omri374 helped me design it :)

File: presidio-image-redactor/Pipfile
Comment (niwilso) [inline presidio-image-redactor/Pipfile:15]: With the additional of OpenCV, please check the license and add it to https://github.com/microsoft/presidio/blob/main/NOTICE if available for commercial use.

Comment (niwilso) [reply presidio-image-redactor/Pipfile:15]: Also, I believe you should update the pip lock file as well.

Comment (omri374) [reply presidio-image-redactor/Pipfile:15]: Agree, NOTICE should be updated. We removed pipfile.lock from the different packages due to the different issues it caused.

**Review by omri374 — DISMISSED** [review]
Looks great! Added a few minor comments

File: presidio-image-redactor/presidio_image_redactor/image_processing_engine.py
Comment (niwilso) [reply presidio-image-redactor/presidio_image_redactor/image_processing_engine.py]: **nit**: Typo/formatting issue with the newly added description.

**Review by niwilso — APPROVED** [review]
Looks good to me now! Will re-approve as needed if code changes are made to address the failing e2e tests.

**Review by niwilso — DISMISSED** [review]

File: presidio-image-redactor/presidio_image_redactor/image_processing_engine.py
Comment (niwilso) [reply presidio-image-redactor/presidio_image_redactor/image_processing_engine.py:23]: I believe we should be able to just use PIL's standard `Image` type.

@ayabel - Would you mind replacing all `PngImagePlugin.PngImageFile` with just `Image.Image` and seeing if that works? If it does, please update the docstrings to indicate PIL image type rather than the specific PNG type.

File: presidio-image-redactor/presidio_image_redactor/dicom_image_redactor_engine.py
Comment (niwilso) [inline presidio-image-redactor/presidio_image_redactor/dicom_image_redactor_engine.py:531]: **nit**: PNG -> PIL.

**Review by niwilso — APPROVED** [review]
Minor comments but otherwise good to go

**Review by niwilso — DISMISSED** [review]

**Review by omri374 — DISMISSED** [review]

**Review by omri374 — APPROVED** [review]

**Review by omri374 — APPROVED** [review]

**Pull request merged**
